samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,255 @@
+import boto3
+import re
+import logging
+import jinja2
+from typing import *
+
+LOG = logging.getLogger(__name__)
+
+
+def _extract_api_id_from_arn(api_arn: str) -> str:
+    """
+    Extracts an api id from an HTTP or REST api arn.
+
+    NOTE: https://docs.aws.amazon.com/apigateway/latest/developerguide/arn-format-reference.html
+
+    REST API Arn: `arn:partition:apigateway:region::/restapis/api-id`
+
+    HTTP API Arn: `arn:partition:apigateway:region::/apis/api-id`
+
+    Parameters
+    ----------
+    api_arn : str
+        The arn of the HTTP or REST api from which the api id is extracted.
+
+    Returns
+    -------
+    str
+        The api id from the api arn.
+    """
+
+    return api_arn[api_arn.rindex("/") + 1 :]
+
+
+def _get_permissions_map() -> dict:
+    """
+    Returns a dictionary mapping a Regular Expression representing the arn of a specific resource to the AWS Resource Type and the default permissions generated for that resource.
+
+    Returns
+    -------
+    default_permissions_map : dict
+        A dictionary mapping a Regular Expression representing the arn of a resource to an object containing the AWS Resource Type and the list of IAM Action permissions generated for that resource.
+    """
+
+    partition_ex = r"(aws|aws-cn|aws-us-gov)"
+    region_ex = r"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d"
+    account_ex = r"\d{12}"
+
+    default_permissions_map = {
+        rf"^arn:{partition_ex}:lambda:{region_ex}:{account_ex}:function:[\w-]+(:\d+)?$": {
+            "Resource": "AWS::Lambda::Function",
```

Comment:
Will this give permission to execute all lambda functions in the same account? If so is there a way to limit this permission to the single lambda resource?

Reply:
So `"AWS::Lambda::Function"` is just a placeholder. If that is the case, can we use a different value here since using type as placeholder can lead to misunderstanding as if we are using the type value here?

Reply:
The reason the types are there are at all is because APIGW IAM Statements have their `Resource` field as [something other than the API arn](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html), so they require special logic.

When we match an arn to a regular expression, I thought it would be more clear to denote what resource type the ARN is supposed to match to. That way, we can more clearly detect when an ARN is a APIGW RestApi arn, or something else, by checking the `Resource` field of the matched object.

I get why this is confusing, because the information is really only used in the APIGW cases. I think a better refactor would be to just have a list of constants with clear names as the regular expressions.

Eg. `LAMBDA_FUNCTION_ARN_REGEX` = `....` and `APIGW_RESTAPI_REGEX`. Then, we when checking for a match, we can say "if the expression matched was `APIGW_RESTAPI_REGEX`, do this logic.